TwitchDownloaderCore/TwitchHelper.cs

```diff
@@ -162,39 +164,67 @@ public static async Task<EmoteResponse> GetThirdPartyEmoteData(string streamerId
                     string name = emote["code"].ToString();
                     string mime = emote["imageType"].ToString();
                     string url = String.Format("https://cdn.betterttv.net/frankerfacez_emote/{0}/[scale]", id);
-                    data.FFZ.Add(new EmoteResponseItem() { Id = id, Code = name, ImageType = mime, ImageUrl = url });
+                    emoteReponse.FFZ.Add(new EmoteResponseItem() { Id = id, Code = name, ImageType = mime, ImageUrl = url });
                 }
             }
 
             if (getStv)
             {
-                JArray STV = JArray.Parse(await httpClient.GetStringAsync("https://7tv.io/v3/emote-sets/global"));
+                string globalEmoteResponse = await httpClient.GetStringAsync("https://7tv.io/v3/emote-sets/global");
+                JsonElement globalEmoteElement = JsonDocument.Parse(globalEmoteResponse).RootElement.GetProperty("emotes");
+                JArray stvEmotes = JArray.Parse(globalEmoteElement.ToString());
 
                 if (streamerId != null)
                 {
                     //Channel might not have 7TV emotes
                     try
                     {
-                        STV.Merge(JArray.Parse(await httpClient.GetStringAsync(String.Format("https://7tv.io/v3/users/twitch/{0}", streamerId))));
+                        string streamerEmoteResponse = await httpClient.GetStringAsync(string.Format("https://7tv.io/v3/users/twitch/{0}", streamerId));
+                        JsonElement streamerEmoteSetElement = JsonDocument.Parse(streamerEmoteResponse).RootElement.GetProperty("emote_set");
+                        JArray streamerEmoteArray = JArray.Parse(streamerEmoteSetElement.GetProperty("emotes").ToString());
+                        stvEmotes.Merge(streamerEmoteArray);
                     }
                     catch { }
                 }
 
-
-                foreach (var emote in STV)
+                foreach (var stvEmote in stvEmotes)
                 {
-                    string id = emote["id"].ToString();
-                    string name = emote["name"].ToString();
-                    string mime = emote["mime"].ToString().Split('/')[1];
-                    string url = String.Format("https://cdn.7tv.app/emote/{0}/[scale]x", id);
-                    EmoteResponseItem emoteResponse = new EmoteResponseItem() { Id = id, Code = name, ImageType = mime, ImageUrl = url };
-                    if (emote["visibility_simple"].ToList().Contains("ZERO_WIDTH"))
+                    string emoteId = stvEmote["id"].ToString();
+                    string emoteName = stvEmote["name"].ToString();
+                    JObject emoteData = (JObject)stvEmote["data"];
+                    JObject emoteHost = (JObject)emoteData["host"];
+                    JArray emoteFiles = JArray.Parse(emoteHost["files"].ToString());
```

Comment:
Probably don't need to convert back, probably just directly cast it to (JArray) as you have for JObject already.
https://www.newtonsoft.com/json/help/html/QueryJson.htm

Reply:
When I use a `JArray` cast, the breakpoint debugger exits step mode when it hits that line, so I'd rather keep it as `JArray.Parse` so breakpoints still work after the statement